packaging/leapp.spec

```diff
@@ -41,67 +48,79 @@ URL:        https://oamg.github.io/leapp/
 BuildArch:  noarch
 
+Requires: %{leapp_python_name}-%{name} = %{version}-%{release}
+
 %if !0%{?fedora}
-%if %{with python3}
-Requires: python3-%{name} = %{version}-%{release}
-%else
-Requires: python2-%{name} = %{version}-%{release}
-%endif
+# TODO: We plan to drop the dependency once we get ack for the planned changes.
+# # In such a case, leapp will not install leapp-repository automatically. For
+# # the full installation, people will have to install a metapackage which will
+# # be part of a particular leapp-repository (e.g. leapp-upgrade metapackage)
+# # Currently the leapp utility does not provide any subcommands (but help)
+# # so it doesn't require the leapp-repository anymore.
 
 # Just ensure the leapp repository will be installed as well. Compatibility
 # should be specified by the leapp-repository itself
 Requires: leapp-repository
 %endif # !fedora
 
+# FIXME: update the description
+# NOTE: tha man page is not updated yet!
 %description
-Leapp tool for handling upgrades.
+Leapp utility provides the possibility to use the Leapp framework via CLI.
+The utility itself does not define any subcommands but "help". All leapp
+subcommands are expected to be provided by other packages under a specific
+directory. See the man page for more details.
 
 
 ##################################################
 # snactor package
 ##################################################
 %package -n snactor
 Summary: %{summary}
-%if %{with python3}
-Requires: python3-%{name} = %{version}-%{release}
-%else
-Requires: python2-%{name} = %{version}-%{release}
-%endif
+Requires: %{leapp_python_name}-%{name} = %{version}-%{release}
 
 %description -n snactor
 Leapp's snactor tool - actor development environment utility for creating and
 managing actor projects.
 
 ##################################################
-# Python 2 library package
+# the library package (the framework itself)
 ##################################################
-%if %{with python2}
-
-%package -n python2-%{name}
+%package -n %{leapp_python_name}-%{name}
 
 Summary: %{summary}
-%{?python_provide:%python_provide python2-%{name}}
+%{?python_provide:%python_provide %{leapp_python_name}-%{name}}
 
-%if 0%{?rhel} && 0%{?rhel} == 7
-# RHEL 7
+%if %{leapp_python} == 2
+# RHEL 7 only
 BuildRequires:  python-devel
 BuildRequires:  python-setuptools
-%else # rhel <> 7 or fedora
-BuildRequires:  python2-devel
-BuildRequires:  python2-setuptools
-%endif # rhel <> 7
+Conflicts:      python3-%{name}
+%else
+BuildRequires:  python3-devel
+BuildRequires:  python3-setuptools
+Conflicts:      python2-%{name}
+
+# NOTE:
+# comment out the python abi requirement. we need to discuss this one later.
+# currently, with the generated python(abi) = 3.6 it's not possible to preserve
+# the rpm on RHEL 9 system. let's talk about consequences of the removal later
+# and how to deal with possible problems
+# %%{?system_python_abi}
+# %%{?python_disable_dependency_generator}
```

Comment:
keeping the whole note with commented out macros to make sure people are aware why we are not using them (currently not sure whether we need or not the second one, I will check it later. it can be removed or allowed in future if we discover it's really (not) needed.

Reply:
dropping the note. I've probably realized what the poem was.